### PR TITLE
ci: Update local host pipeline to use all in one deployment

### LINKED ISSUE #878: Update pipeline ArmoniK: use all-in-one deployment
- [x]  Create a tag for partitions 

- [x]  Create a function that allows you to manage the partition flow by test

- [x]   Integrate into the all_in_one pipeline

### PR DESCRIPTION
**Pull request by iadjadj**
fix #878

### CODE REVIEW

**Review by aneojgurhem — APPROVED** [review]

**Review by lemaitre-aneo — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -135,80 +98,58 @@ jobs:
...
+           docker run --rm \
+             -e GrpcClient__Endpoint="${CONTROL_PLANE_URL}" \
+             -e Partition="stream" \
+             dockerhubaneo/armonik_core_stream_test_client:${{ steps.core.outputs.tag }}
       - name: Show logs
         if: always()
         run: |
           set -e
           mkdir -p core-stream-without-ingress/infra/generated \
             core-stream-without-ingress/infra/tfstates \
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:149]: Rename "core-stream-without-ingress" by "core-stream"

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -135,80 +98,58 @@ jobs:
...
+           export CONTROL_PLANE_URL=$(cat infrastructure/quick-deploy/localhost/all/generated/armonik-output.json | jq -r '.armonik.control_plane_url')
+           docker run --rm \
+             -e GrpcClient__Endpoint="${CONTROL_PLANE_URL}" \
+             -e Partition="stream" \
+             dockerhubaneo/armonik_core_stream_test_client:${{ steps.core.outputs.tag }}
       - name: Show logs
         if: always()
         run: |
           set -e
           mkdir -p core-stream-without-ingress/infra/generated \
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:148]: Rename "core-stream-without-ingress" by "core-stream"

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -135,80 +98,58 @@ jobs:
...
             core-stream-without-ingress/infra/tfstates \
             core-stream-without-ingress/app
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:150]: Rename "core-stream-without-ingress/app" by "core-stream/app"

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -135,80 +98,58 @@ jobs:
...
-            infrastructure/quick-deploy/localhost/storage/generated/storage-output.json \
-            infrastructure/quick-deploy/localhost/monitoring/generated/monitoring-output.json \
-            infrastructure/quick-deploy/localhost/armonik/generated/armonik-output.json \
-            core-stream-without-ingress/infra/generated
-          cp infrastructure/quick-deploy/localhost/keda/generated/backend/keda-terraform.tfstate \
-            infrastructure/quick-deploy/localhost/storage/generated/backend/storage-terraform.tfstate \
-            infrastructure/quick-deploy/localhost/monitoring/generated/backend/monitoring-terraform.tfstate \
-            infrastructure/quick-deploy/localhost/armonik/generated/backend/armonik-terraform.tfstate \
-            core-stream-without-ingress/infra/tfstates
+          cp infrastructure/quick-deploy/localhost/all/generated/armonik-output.json  core-stream-without-ingress/infra/generated
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:151]: Replace "core-stream-without-ingress/infra/generated" by "core-stream/infra/generated"

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -135,80 +98,58 @@ jobs:
...
+          cp infrastructure/quick-deploy/localhost/all/generated/armonik-terraform.tfstate core-stream-without-ingress/infra/tfstates
           sudo cp -rL /var/log/pods/armonik_* core-stream-without-ingress/app
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:153]: replace "core-stream-without-ingress/app" by "core-stream/app"

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -217,419 +158,228 @@ jobs:
...
-            -e HtcMock__EnableUseLowMem=true \
-            -e HtcMock__EnableSmallOutput=true \
-            -e HtcMock__EnableFastCompute=true \
-            dockerhubaneo/armonik_core_htcmock_test_client:${{ steps.core.outputs.tag }}
-
+            set -ex
+            export CONTROL_PLANE_URL=$(cat infrastructure/quick-deploy/localhost/all/generated/armonik-output.json | jq -r '.armonik.control_plane_url')
+            docker run --rm \
+              -e GrpcClient__Endpoint="${CONTROL_PLANE_URL}" \
+              -e HtcMock__NTasks=2000 \
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:219]: Decrease the number of tasks to 1000 tasks

File: .github/workflows/deploy.yml [hunk 7]
```diff
@@ -217,419 +158,228 @@ jobs:
...
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:253]: Decrease the number of tasks to 1000 tasks

File: .github/workflows/deploy.yml [hunk 8]
```diff
@@ -217,419 +158,228 @@ jobs:
...
-            infrastructure/quick-deploy/localhost/storage/generated/backend/storage-terraform.tfstate \
-            infrastructure/quick-deploy/localhost/monitoring/generated/backend/monitoring-terraform.tfstate \
-            infrastructure/quick-deploy/localhost/armonik/generated/backend/armonik-terraform.tfstate \
-            htcmock-disabled-parameters-without-ingress/infra/tfstates
-          sudo cp -rL /var/log/pods/armonik_* htcmock-disabled-parameters-without-ingress/app
-          sudo chown $USER -R htcmock-disabled-parameters-without-ingress
-          tar -czf htcmock-disabled-parameters-without-ingress.tar.gz htcmock-disabled-parameters-without-ingress
-          aws s3 cp htcmock-disabled-parameters-without-ingress.tar.gz s3://${{ secrets.AWS_LOG_BUCKET_NAME }}/armonik-pipeline/${{ github.run_number }}/${{ github.run_attempt }}/htcmock-disabled-parameters-without-ingress.tar.gz
+            set -e
+            mkdir -p htcmock-disabled-parameters-without-ingress/infra/generated \
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:267]: Replace "htcmock-disabled-parameters-without-ingress/infra/generated " by "htcmock/infra/generated "

File: .github/workflows/deploy.yml [hunk 9]
```diff
@@ -217,419 +158,228 @@ jobs:
...
+              htcmock-disabled-parameters-without-ingress/infra/tfstates \
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:268]: Replace "htcmock-disabled-parameters-without-ingress/infra/tfstates " by "htcmock/infra/tfstates "

File: .github/workflows/deploy.yml [hunk 10]
```diff
@@ -217,419 +158,228 @@ jobs:
...
+              htcmock-disabled-parameters-without-ingress/app
+            cp infrastructure/quick-deploy/localhost/all/generated/armonik-output.json  htcmock-disabled-parameters-without-ingress/infra/generated
+            cp infrastructure/quick-deploy/localhost/all/generated/armonik-terraform.tfstate  htcmock-disabled-parameters-without-ingress/infra/tfstates
+            sudo cp -rL /var/log/pods/armonik_* htcmock-disabled-parameters-without-ingress/app
+            sudo chown $USER -R htcmock-disabled-parameters-without-ingress
+            tar -czf htcmock-disabled-parameters-without-ingress.tar.gz htcmock-disabled-parameters-without-ingress
+            aws s3 cp htcmock-disabled-parameters-without-ingress.tar.gz s3://${{ secrets.AWS_LOG_BUCKET_NAME }}/armonik-pipeline/${{ github.run_number }}/${{ github.run_attempt }}/htcmock-disabled-parameters-without-ingress.tar.gz
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:275]: Replace "htcmock-disabled-parameters-without-ingress " by "htcmock "

File: .github/workflows/deploy.yml [hunk 11]
```diff
@@ -217,419 +158,228 @@ jobs:
...
-          sudo chown $USER -R htcmock-without-tls-without-mtls
-          tar -czf htcmock-without-tls-without-mtls.tar.gz htcmock-without-tls-without-mtls
-          aws s3 cp htcmock-without-tls-without-mtls.tar.gz s3://${{ secrets.AWS_LOG_BUCKET_NAME }}/armonik-pipeline/${{ github.run_number }}/${{ github.run_attempt }}/htcmock-without-tls-without-mtls.tar.gz
-        env:
-          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          AWS_EC2_METADATA_DISABLED: true
+               type: localhost
 
   benchWithoutIngress:
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:285]: Replace "benchWithoutIngress" by "bench"

File: .github/workflows/deploy.yml [hunk 12]
```diff
@@ -217,419 +158,228 @@ jobs:
...
+             set -e
+             mkdir -p bench-without-ingress/infra/generated \
+               bench-without-ingress/infra/tfstates \
+               bench-without-ingress/app
+             cp infrastructure/quick-deploy/localhost/all/generated/armonik-output.json bench-without-ingress/infra/generated
+             cp infrastructure/quick-deploy/localhost/all/generated/armonik-terraform.tfstate bench-without-ingress/infra/tfstates
+             sudo cp -rL /var/log/pods/armonik_* bench-without-ingress/app
+             sudo chown $USER -R bench-without-ingress
+             tar -czf bench-without-ingress.tar.gz bench-without-ingress
+             aws s3 cp bench-without-ingress.tar.gz s3://${{ secrets.AWS_LOG_BUCKET_NAME }}/armonik-pipeline/${{ github.run_number }}/${{ github.run_attempt }}/bench-without-ingress.tar.gz
```
Comment (lzianekhodja) [inline .github/workflows/deploy.yml:376]: Replace "bench-without-ingress" by "bench"

**Review by lzianekhodja — APPROVED** [review]

**Review by lzianekhodja — APPROVED** [review]

**Pull request merged**